engine/consensus/ingestion/engine.go

```diff
@@ -131,17 +140,33 @@ func (e *Engine) process(originID flow.Identifier, event interface{}) error {
 	switch ev := event.(type) {
 	case *flow.CollectionGuarantee:
 		e.metrics.MessageReceived(metrics.EngineConsensusIngestion, metrics.MessageCollectionGuarantee)
-		return e.onGuarantee(originID, ev)
+		err := e.onGuarantee(originID, ev)
+		if err != nil {
+			if engine.IsInvalidInputError(err) {
+				e.log.Error().Str("origin", originID.String()).Err(err).Msg("received invalid collection guarantee")
+				return nil
+			}
+			if engine.IsOutdatedInputError(err) {
+				e.log.Warn().Str("origin", originID.String()).Err(err).Msg("received outdated collection guarantee")
+				return nil
+			}
+			if engine.IsUnverifiableInputError(err) {
+				e.log.Warn().Str("origin", originID.String()).Err(err).Msg("received unverifiable collection guarantee")
+				return nil
+			}
```

Comment:
I'm afraid this three `ifs` will be duplicated in a lot of other places.

Can we combine it with engine.LogError?

Seems the only difference here is to populate the unknown error in the end instead of logging it.

Reply:
Being explicit here (at the cost of duplicating 12 lines of code) was specifically my intention: if we receive an error other than the expected, this implementation here will crash. While the `engine.LogError` does not treat the error and just logs it. Furthermore, if some other engineer adds another "expected error" to the `engine.LogError`, we would also ignore it here (even though in this context, it is unexpected and should be fatal). 

My preference would be to be explicit for each engine. For example, when engines are processing locally _trusted_ messages, an `IsInvalidInputError` should certainly be fatal. I am happy if we implement for _each engine a dedicated _`LogError` functionality. But I am strongly opposed to the one-pattern-fits-all-engines approach as implemented by the `engine.LogError`. I think 12 lines of code duplication per engine is acceptable.

Reply:
Cool, makes sense.